packages/wallets/src/features/cashier/screens/WithdrawalErrorScreen/WithdrawalErrorScreen.tsx

```diff
@@ -1,16 +1,41 @@
 import React from 'react';
+import classNames from 'classnames';
 import { TSocketError } from '@deriv/api-v2/types';
 import { WalletsErrorScreen } from '../../../../components';
 import { CryptoWithdrawalErrorCodes } from '../../../../constants/errorCodes';
 
 type TProps = {
-    error: TSocketError<'cashier'>['error'];
+    currency?: string;
+    error?: TSocketError<'cashier'>['error'];
     resetError: VoidFunction;
     setResendEmail: React.Dispatch<React.SetStateAction<boolean>>;
 };
 
-const WithdrawalErrorScreen: React.FC<TProps> = ({ error, resetError, setResendEmail }) => {
-    if (error.code === CryptoWithdrawalErrorCodes.InvalidToken) {
+const WithdrawalErrorScreen: React.FC<TProps> = ({ currency, error, resetError, setResendEmail }) => {
+    const CryptoConnectionError = error?.code === CryptoWithdrawalErrorCodes.CryptoConnectionError;
+    const CryptoInvalidAddress = error?.code === CryptoWithdrawalErrorCodes.CryptoInvalidAddress;
+    const InvalidToken = error?.code === CryptoWithdrawalErrorCodes.InvalidToken;
+    const SuspendedCurrencyWithdrawal =
+        error?.code ===
+        (CryptoWithdrawalErrorCodes.SuspendedCurrency || CryptoWithdrawalErrorCodes.SuspendedWithdrawal);
+
+    const getErrorTitle = () => {
+        if (CryptoInvalidAddress) return 'Error';
+        if (SuspendedCurrencyWithdrawal) return `${currency} Wallet withdrawals are temporarily unavailable`;
+        if (CryptoConnectionError) return 'Maintenance in progess';
+        return undefined;
+    };
+
+    const getErrorMessage = () => {
+        if (SuspendedCurrencyWithdrawal)
+            return `Due to system maintenance, withdrawals with your ${currency} Wallet are unavailable at the moment. Please try again later.`;
+        return error?.message;
+    };
+
+    const errorTitle = error ? getErrorTitle() : undefined;
+    const errorMessage = error ? getErrorMessage() : undefined;
```

Comment:
what about:

```
const getErrorTitle = (): string | undefined => {
        switch (error?.code) {
            case CryptoWithdrawalErrorCodes.CryptoInvalidAddress:
                return 'Error';
            case CryptoWithdrawalErrorCodes.SuspendedCurrency:
            case CryptoWithdrawalErrorCodes.SuspendedWithdrawal:
                return `${currency} Wallet withdrawals are temporarily unavailable`;
            case CryptoWithdrawalErrorCodes.CryptoConnectionError:
                return 'Maintenance in progress';
            default:
                return undefined;
        }
    };

    const getErrorMessage = (): string => {
        if (error?.code === CryptoWithdrawalErrorCodes.SuspendedCurrency || error?.code === CryptoWithdrawalErrorCodes.SuspendedWithdrawal) {
            return `Due to system maintenance, withdrawals with your ${currency} Wallet are unavailable at the moment. Please try again later.`;
        }
        return error?.message || 'An unexpected error occurred.';
    };

    const errorTitle = getErrorTitle();
    const errorMessage = getErrorMessage();
    
    ```
    
    basically to encapsulte logic of getting error message into one function with all its logic, and then encapsulte the rror message logic in anothre function,

Reply:
basically to encapsulte logic of getting error message into one function with all its logic, and then encapsulte the rror message logic in anothre function, to make it clear and readable.

Reply:
as indeed, as per other comments, its becoming quite hard to read, but if you only extract the getErrorMessage() and getErrorTitle() function above the component, it will immediately became way cleaner with little effort

Reply:
Updated the implementation to make it more readable, please rereview 🙏